### DISCUSSION TITLE
I got iperf 3.11 to compile in windows with clang 14

### DISCUSSION THREAD

**Discussion by devilsclaw** [esnet/iperf#1404] [Q&A]
I have a project where I am building a GUI interface with iperf 3 and I did not want to deal with running the program and parsing data.

I used a template of someones else cmake build files and modified them to properly check what is available in windows. I then patched Iperf to compile in windows.

There are a couple things not working in windows like becoming a daemon, TCP_MAXSEG unless running in windows 10 or above, unlink, creating pid file and kill, termios stuff, I currently have SSL disabled since I don't need it but it would not be hard to get working. I use rand_s for the cookies since there is no urandom in windows, cpu util is not working, get sys info not working, setnonblocking not working.

I have tested it and it compiles and I am able to run throughput test.

I am not sure if there is interest in getting this working in windows natively without cygwin or not, but I have something that does for the most part if there is interest.

**Comment by devilsclaw** [comment]
It also works in linux since that is my primary operating system

**Reply by BlackDogWhite** [reply]
Would you be willing to give me a copy of the compiled code?  Either your GUI or the cmd line utility?  I want to try out some testing but don't want to take the time to build for Windows.

**Reply by devilsclaw** [reply]
Here is the non GUI version. if your just looking for a windows version of iperf there are other people who have done it and I think not as limited as this one. but if your looking for a prof of concept then this is it.
https://drive.google.com/file/d/10-KfQ6dnEiVNc5eXYK6HF2eRwknNYCgO/view?usp=sharing
